### PR TITLE
Add more sytem tests for xnet socket API

### PR DESCRIPTION
**Pull request by astarche**
### What does this Pull Request accomplish?

Add more sytem tests for xnet socket API.

### Why should this Pull Request be merged?

Fixes [AB#1891157](https://ni.visualstudio.com/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/1891157).

### What testing has been done?

Ran and passed all Xnet tests.

### CODE REVIEW

**Comment by astarche** [comment]
image

File: source/tests/system/nixnetsocket_driver_api_tests.cpp [hunk 1]
```diff
@@ -495,19 +495,149 @@ TEST_F(NiXnetSocketLoopbackTests, OpenedStackThenCloseOriginal_GetStackInfoFromO
...
+  EXPECT_SUCCESS(close_accepted);
+  EXPECT_XNET_ERROR(GENERIC_NXSOCKET_ERROR, INVALID_SOCKET_ERROR, INVALID_SOCKET_MESSAGE, client::get_sock_name(stub(), tx_socket.socket()));
+  EXPECT_XNET_ERROR(GENERIC_NXSOCKET_ERROR, INVALID_SOCKET_ERROR, INVALID_SOCKET_MESSAGE, client::get_sock_name(stub(), rx_socket.socket()));
+  EXPECT_XNET_ERROR(GENERIC_NXSOCKET_ERROR, INVALID_SOCKET_ERROR, INVALID_SOCKET_MESSAGE, client::get_sock_name(stub(), accept.socket()));
+}
+
+TEST_F(NiXnetSocketLoopbackTests, UdpSocketsOnWiredPhysicalLoopback_SendAndReceiveData_ReceivesExpectedData)
+{
+  const auto MESSAGE = "HelloWorld"s;
+  const auto rx_stack = client::ip_stack_create(stub(), "", create_simple_config("ENET1", "10.0.0.1"));
```
Comment (reckenro) [inline source/tests/system/nixnetsocket_driver_api_tests.cpp:598]: Nit: Consider adding `RX_ADDR`/`TX_ADDR` constants to use here and in the lookup calls.

**Review by reckenro — APPROVED** [review]

File: source/tests/system/nixnetsocket_driver_api_tests.cpp [hunk 2]
```diff
@@ -495,19 +495,149 @@ TEST_F(NiXnetSocketLoopbackTests, OpenedStackThenCloseOriginal_GetStackInfoFromO
...
+  EXPECT_SUCCESS(listen);
+  EXPECT_SUCCESS(tx_socket);
+
+  const auto connect = client::connect(stub(), tx_socket.socket(), get_name.addr());
+  const auto accept = client::accept(stub(), rx_socket.socket());
+  const auto send = client::send(stub(), tx_socket.socket(), MESSAGE, 0);
+  const auto recv = client::recv(stub(), accept.socket(), MESSAGE.size(), 0);
+
+  EXPECT_SUCCESS(connect);
+  EXPECT_SUCCESS_WITH_STATUS(send, MESSAGE.size());
```
Comment (mgumble) [inline source/tests/system/nixnetsocket_driver_api_tests.cpp:560]: nit: if you could flip the macro so it is expected, actual must of missed that in a previous PR

File: source/tests/system/nixnetsocket_driver_api_tests.cpp [hunk 3]
```diff
@@ -495,19 +495,149 @@ TEST_F(NiXnetSocketLoopbackTests, OpenedStackThenCloseOriginal_GetStackInfoFromO
...
+  const auto wait = client::ip_stack_wait_for_interface(stub(), rx_stack.stack_ref(), "", 5000);
+  const auto rx_socket = socket(stub(), rx_stack.stack_ref());
+  auto sock_addr = lookup_ipv4_sock_addr(stub(), rx_stack.stack_ref(), RX_ADDR, 0);
+  const auto bind = client::bind(stub(), rx_socket.socket(), sock_addr);
+  const auto get_name = client::get_sock_name(stub(), rx_socket.socket());
+  const auto listen = client::listen(stub(), rx_socket.socket(), 10);
+  const auto tx_socket = socket(stub(), tx_stack.stack_ref());
+  EXPECT_SUCCESS(tx_stack);
+  EXPECT_SUCCESS(rx_stack);
+  EXPECT_SUCCESS(wait);
```
Comment (mgumble) [inline source/tests/system/nixnetsocket_driver_api_tests.cpp:547]: Info: Looks like github visualizes "wait" as a special word.

**Review by mgumble — APPROVED** [review]

**Pull request merged**